iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/json/ResourceDeclarationImplTest.java

```diff
@@ -256,4 +257,45 @@ void shouldParseResourceWithComplexProperties() {
     assertThat(property.key().value()).isEqualTo("sourceAddressPrefixes");
     assertThat(property.value()).hasKind(ARRAY_EXPRESSION).hasArrayExpressionValues("0.0.0.0/0");
   }
+
+  @Test
+  void shouldParseResourceWithChildResourcesInIt() {
+    String code = code("{",
+      "  \"resources\": [",
+      "    {",
+      "      \"name\": \"parent resource\",",
+      "      \"type\": \"Microsoft.Network/networkSecurityGroups\",",
+      "      \"apiVersion\": \"2022-11-01\",",
+      "      \"resources\": [",
+      "        {",
+      "          \"name\": \"child resource\",",
+      "          \"type\": \"securityRules\",",
+      "          \"apiVersion\": \"2022-11-01\",",
+      "          \"properties\": {\"attr\": \"value\"}",
+      "        }",
+      "      ]",
+      "    }",
+      "  ]",
+      "}");
+
+    File tree = (File) parser.parse(code, null);
+    assertThat(tree.statements()).hasSize(1);
```

Comment:
I would suggest moving this assertion at the end. In case of some errors, the next assertions may catch it. 
In case there are more statements, we will know that at least the first is as expected.